pandas/tests/test_categorical.py

```diff
@@ -3822,6 +3822,34 @@ def test_cat_equality(self):
         pytest.raises(TypeError, lambda: a > b)
         pytest.raises(TypeError, lambda: b > a)
 
+    def test_unordered_different_order_equal(self):
+        # https://github.com/pandas-dev/pandas/issues/16014
+        c1 = Categorical(['a', 'b'], categories=['a', 'b'], ordered=False)
+        c2 = Categorical(['a', 'b'], categories=['b', 'a'], ordered=False)
```

Comment:
A lot of the tests above are with Series instead of Categorical. 
Don't need to change all, but maybe good to test this with a Series as well